src/frontend/src/common/components/library/data_subview/components/EditSamplesConfirmationModal/index.tsx

```diff
@@ -32,6 +41,35 @@ const EditSamplesConfirmationModal = ({
   onClose,
   open,
 }: Props): JSX.Element | null => {
+  const [isValid, setIsValid] = useState(false);
+  const [metadata, setMetadata] =
+    useState<SampleIdToEditMetadataWebform | null>(null);
+
+  const { data: namedLocationsData } = useNamedLocations();
+  const namedLocations =
+    namedLocationsData?.namedLocations || ([] as NamedGisaidLocation[]);
```

Comment:
```suggestion
  const namedLocations: NamedGisaidLocation[] =
    namedLocationsData?.namedLocations ?? [];
```

When you use `as` in TS, you basically tell the compiler "I'm sure this thing is the right type". Although that is the case here, I'd like us to be in the habit of limiting the use of `as` wherever possible because it actively diminishes type safety.

Reply:
https://stackoverflow.com/questions/47994926/detailed-differences-between-type-annotation-variable-type-and-type-assertion

Reply:
got it, thanks for the explanation, i'll try to limit my use of `as` going forwards unless necessary